pemFioi/quickpi/blocklyQuickPi_lib.js

```diff
@@ -2931,18 +2931,28 @@ var getContext = function (display, infos, curLevel) {
     context.resetSensors = function() {
         for (var iSensor = 0; iSensor < infos.quickPiSensors.length; iSensor++) {
             var sensor = infos.quickPiSensors[iSensor];
-
-            sensor.state = null;
-            sensor.screenDrawing = null;
-            sensor.lastDrawnTime = 0;
-            sensor.lastDrawnState = null;
-            sensor.callsInTimeSlot = 0;
-            sensor.lastTimeIncrease = 0;
-            sensor.removed = false;
-            sensor.quickStore = null;
-
+            if (context.sensorsSaved[sensor.name]) {
```

Comment:
You should restore the saved state only when in "user mode" (`context.autoGrading = false;` if i'm not forgetting something).

Reply:
Thank you, I forgot about this one